src/components/wrappers/PopUp.vue

```diff
@@ -0,0 +1,131 @@
+<script>
+import ButtonSign from '@/components/formElements/ButtonSign.vue';
+
+export default {
+  components: {
+    ButtonSign,
+  },
+  props: {
+    informational: Boolean,
+    stopPropagation: {
+      type: Boolean,
+      default: false,
+    },
+    buttonStyleType: {
+      type: String,
+      default: 'hint',
+    },
+    position: {
+      type: String,
+      default: 'right',
+    },
+  },
+  emits: ['close'],
+  data() {
+    return {
+      isShown: false,
+    };
+  },
+  methods: {
+    validatePopUpClosing(event) {
+      const target = document.querySelector('.pop-up-content');
+
+      if (event.target !== target && !target.contains(event.target)) {
+        this.closePopUp();
+      }
+    },
+    openPopUp() {
+      this.isShown = true;
+      document.addEventListener('click', this.validatePopUpClosing);
+    },
+    closePopUp() {
+      this.isShown = false;
+      document.removeEventListener('click', this.validatePopUpClosing);
```

Comment:
I don't see beforeDestroy removing this listener

Reply:
the component PopUp is not destoyed (unmounted) when the pop-up is closed because the toggling button responsible for pop-up showing is inside the component as well

so there is no any sense to remove event listener in this hook, it is removed when pop-up is closed

Reply:
but it could be destroyed when it's open when you are destroying parent component

Reply:
for sure, I agree

however, as far as I am concerned, if DOM element which has no other references is removed, its listeners are also removed from memory in JS
(I tested it as well)

so, as for me, removing the listener separately in beforeUnmount is redundant since it appears duplicating browser's natural behavior

nevertheless, if you insist that it is a necessary step to make sure that the listener is definitely removed, I will add